extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionValidateOnMigrateTest.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.flyway.test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import javax.inject.Inject;
+
+import org.flywaydb.core.Flyway;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.agroal.api.AgroalDataSource;
+import io.quarkus.test.QuarkusUnitTest;
+
+/**
+ * Test suite for Flyway's validate.
+ * <p>
+ * Assures Flyway's validate runs successfully after SQL migrations are applied, by checking that
+ * they do match the one(s) available locally.
+ */
+public class FlywayExtensionValidateOnMigrateTest {
```

Comment:
There is no need to test Flyway (they have their own tests) :)

Reply:
This is actually not testing `Flyway` (in my opinion) but the integration with Quarkus: `quarkus.flyway.validate-on-migrate`, which I defaulted to run all the time (read: `true`). The call to `flyway.validate()` in `testFlywayConfigInjection` can be actually deleted since everything happens when the application starts.